internal/pkg/manifest/transform.go

```diff
@@ -0,0 +1,208 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"reflect"
+
+	"github.com/imdario/mergo"
+)
+
+var basicKinds = []reflect.Kind{
+	reflect.Bool,
+	reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
+	reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64,
+	reflect.Float32, reflect.Float64,
+	reflect.Complex64, reflect.Complex128,
+	reflect.Array, reflect.String, reflect.Slice,
+}
+
+type workloadTransformer struct{}
+
+// Transformer returns custom merge logic for workload's fields.
+func (t workloadTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
+	if transform := transformBasic(typ); transform != nil {
+		return transform
+	}
+
+	if typ == reflect.TypeOf(Image{}) {
+		return transformImage()
+	}
+
+	if typ.String() == "map[string]manifest.Volume" {
+		return transformMapStringToVolume()
+	}
+	return nil
+}
+
+type volumeTransformer struct{}
+
+// Transformer returns custom merge logic for volume's fields.
+func (t volumeTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
+	return transformBasic(typ)
+}
+
+func transformBasic(typ reflect.Type) func(dst, src reflect.Value) error {
+	if typ.Kind() == reflect.Slice {
+		return transformSlice()
+	}
+
+	if typ.Kind() == reflect.Ptr {
+		for _, k := range basicKinds {
+			if typ.Elem().Kind() == k {
+				return transformPBasic()
+			}
+		}
+
+		if typ.Elem().Kind() == reflect.Struct {
+			return transformPStruct()
+		}
+	}
+	return nil
+}
+
+func transformSlice() func(dst, src reflect.Value) error {
+	return func(dst, src reflect.Value) error {
+		dst.Set(src)
+		return nil
+	}
+}
+
+func transformMapStringToVolume() func(dst, src reflect.Value) error {
+	return func(dst, src reflect.Value) error {
+		for _, key := range src.MapKeys() {
+			srcElement := src.MapIndex(key)
+			if !srcElement.IsValid() {
+				continue
+			}
+			dstElement := dst.MapIndex(key)
+
+			// Perform default merge
+			dstV := dstElement.Interface().(Volume)
+			srcV := srcElement.Interface().(Volume)
+			err := mergo.Merge(&dstV, srcV, mergo.WithOverride, mergo.WithTransformers(volumeTransformer{}))
+			if err != nil {
+				return err
+			}
+
+			// Set merged value for the key
+			dst.SetMapIndex(key, reflect.ValueOf(dstV))
+		}
+		return nil
+	}
+}
+
+func transformPStruct() func(dst, src reflect.Value) error {
+	return func(dst, src reflect.Value) error {
+		if src.IsNil() {
+			return nil
+		}
+
+		// TODO: these can be removed if we change all pointers struct to struct
```

Comment:
omg this is insane 😱 

I'm okay with merging these changes but we cannot do any release until we get to remove all these patch fixes 🤯